src/overlays/actors/ovl_En_Weiyer/z_en_weiyer.c

```diff
@@ -52,59 +63,590 @@ static ColliderCylinderInit D_80B33A90 = {
+    this->unk_194 = 0;
+    this->actor.speedXZ = 5.0f;
+    this->actionFunc = func_80B32C2C;
+}
+
+void func_80B32494(EnWeiyer* this) {
+    Animation_Change(&this->skelAnime, &D_06000288, 2.0f, 0.0f, 0.0f, 0, -8.0f);
+    this->unk_194 = 40;
+    this->collider.base.atFlags |= AT_ON;
+    this->actionFunc = func_80B32D30;
+}
+
+void func_80B32508(EnWeiyer* this) {
+    this->unk_194 = 200;
+    this->collider.base.atFlags |= AT_ON;
+    this->skelAnime.playSpeed = 3.0f;
+    this->actionFunc = func_80B32E34;
+}
+
+void func_80B32538(EnWeiyer* this) {
+    this->unk_194 = 200;
+    this->unk_196 = this->actor.yawTowardsPlayer + 0x8000;
+    this->unk_27C = this->actor.world.pos.y;
+    this->actor.speedXZ = CLAMP_MAX(this->actor.speedXZ, 4.0f);
+    this->collider.base.atFlags &= ~AT_ON;
+    this->skelAnime.playSpeed = 1.0f;
+    this->actionFunc = func_80B33018;
+}
+
+void func_80B325A0(EnWeiyer* this) {
+    Animation_Change(&this->skelAnime, &D_06000FC0, 2.0f, 0.0f, 0.0f, 0, -3.0f);
+    this->unk_194 = 40;
+    this->collider.base.atFlags &= ~AT_ON;
+    this->collider.base.acFlags &= ~AC_ON;
+    this->actor.gravity = 0.0f;
+    this->actor.velocity.y = 0.0f;
+    this->actor.speedXZ = 3.0f;
+    func_8003426C(&this->actor, 0x4000, 0xC8, 0, 0x28);
+    this->collider.dim.height = sCylinderInit.dim.height;
+    this->actionFunc = func_80B331CC;
+}
+
+void func_80B32660(EnWeiyer* this) {
+    Animation_Change(&this->skelAnime, &D_06000288, 2.0f, 0.0f, 0.0f, 0, -8.0f);
+    this->unk_194 = 80;
+    this->actor.speedXZ = 0.0f;
+    this->actor.velocity.y = 0.0f;
+    this->actor.gravity = -1.0f;
+    this->collider.dim.height = sCylinderInit.dim.height + 0xF;
+    func_8003426C(&this->actor, 0, 0xC8, 0, 0x50);
+    this->collider.base.atFlags &= ~AT_ON;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_FREEZE);
+    this->actionFunc = func_80B333B8;
+}
+
+void func_80B32724(EnWeiyer* this) {
+    Animation_MorphToLoop(&this->skelAnime, &D_06000FC0, -5.0f);
+    this->unk_194 = 20;
+    func_8003426C(&this->actor, 0x4000, 0xC8, 0, 0x28);
+    this->collider.base.atFlags &= ~AT_ON;
+    this->collider.base.acFlags &= ~AC_ON;
+    this->actor.speedXZ = 3.0f;
+    this->actionFunc = func_80B332B4;
+}
+
+void func_80B327B0(EnWeiyer* this) {
+    this->actor.colorFilterParams |= 0x2000;
+    this->actor.speedXZ = 0.0f;
+    this->actor.velocity.y = 0.0f;
+    this->actionFunc = func_80B33338;
+}
+
+void func_80B327D8(EnWeiyer* this) {
+    this->actor.shape.rot.x = -0x2000;
+    this->unk_194 = -1;
+    this->actor.speedXZ = 5.0f;
+    this->actionFunc = func_80B3349C;
+}
+
+void func_80B32804(EnWeiyer* this, GlobalContext* globalCtx) {
+    WaterBox* sp34;
+    UNK_TYPE sp30;
+
+    this->actor.world.pos.y += 0.5f;
+    this->actor.floorHeight = BgCheck_EntityRaycastFloor4(&globalCtx->colCtx, &this->actor.floorPoly, &sp30,
+                                                          &this->actor, &this->actor.world.pos);
```

Comment:
sp30 should be an s32, since BgCheck_EntityRaycastFloor4's third argument is `s32* bgId`, the bgId of the detected poly.